### PR TITLE
Adding vuid 02651, Validate pDepthStencilResolveAttachment

### PR DESCRIPTION
**Pull request by ziga-lunarg**
When creating a renderPass, if pNext chain of VkSubpassDescripton2 contains pDepthStencilResolveAttachment and attachment is not unused, its format features must support VK_FORMAT_FEATURE_DEPTH_STENCIL_ATTACHMENT_BIT

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -10774,6 +10774,21 @@ bool CoreChecks::ValidateDepthStencilResolve(const VkPhysicalDeviceVulkan12Prope
                              function_name, i, resolve->pDepthStencilResolveAttachment->attachment);
         }
 
+        if (resolve_attachment_not_unused) {
+            VkFormat resolve_format = pCreateInfo->pAttachments[resolve->pDepthStencilResolveAttachment->attachment].format;
+            const VkFormatProperties format_properties = GetPDFormatProperties(resolve_format);
+            const VkFormatFeatureFlags tiling_features = format_properties.optimalTilingFeatures;
```
Comment (sfricke-samsung) [inline layers/core_validation.cpp:10780]: So I realize the issue is `VUID-VkSubpassDescriptionDepthStencilResolve-pDepthStencilResolveAttachment-02651` is not properly written in the spec like `VUID-VkSubpassDescription2-pDepthStencilAttachment-02900`

but this should be using `GetPotentialFormatFeatures` or else someone will have some fun edge case using AHB that hits a false positive here

Comment (jeremyg-lunarg) [reply layers/core_validation.cpp:10780]: Good point, hopefully running the unit tests on Android devices will cover this case

**Review by jeremyg-lunarg — CHANGES_REQUESTED** [review]
@sfricke-samsung 's comment needs to be addressed, otherwise LGTM.

File: layers/core_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/core_validation.cpp:10780]: https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/4758 for the record

Comment (oddhack) [reply layers/core_validation.cpp:10780]: This should be fixed in the 1.2.188 spec update.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM
